src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py

```diff
@@ -3588,6 +3588,34 @@ def test_managedappdef_inline(self, resource_group):
         self.cmd('role assignment delete --assignee {upn} --role contributor ')
         self.cmd('ad user delete --id {upn}')
 
+    @ResourceGroupPreparer()
+    def test_managed_app_def_deployment_mode(self):
+        curr_dir = os.path.dirname(os.path.realpath(__file__))
+
+        self.kwargs.update({
+            'upn': self.create_random_name('testuser', 15) + '@azuresdkteam.onmicrosoft.com',
+            'sub': self.get_subscription_id()
+        })
+        user_principal = self.cmd('ad user create --display-name tester123 --password Test123456789 --user-principal-name {upn}').get_output_in_json()
+        time.sleep(15)  # By-design, it takes some time for RBAC system propagated with graph object change
+        principal_id = user_principal['id']
+        with mock.patch('azure.cli.command_modules.role.custom._gen_guid', side_effect=self.create_guid):
+            role_assignment = self.cmd(
+                'role assignment create --assignee {upn} --role contributor --scope "/subscriptions/{sub}" ').get_output_in_json()
```

Comment:
Please add the `role assignment delete` command and `ad user delete` after the test execution is completed, as they do not belong to the resource group, so they will not be automatically deleted after the resource group is deleted
```
        self.cmd('role assignment delete --assignee {upn} --role contributor ')
        self.cmd('ad user delete --id {upn}')
```